include/malloc.h

```diff
@@ -36,6 +38,9 @@ typedef struct malloc_pool {
   uint32_t mp_magic;                /* Detect programmer error. */
   const char *mp_desc;              /* Printable type name. */
   struct ma_list mp_arena;          /* Queue of managed arenas. */
+  mtx_t mutex;                      /* Mutex protecting structure */
```

Comment:
I don't need this `mp_pages_min` value for now since pages are only allocated when needed and never freed. I would rather keep just `mp_pages_used` and `mp_pages_max`.